### PR TITLE
Make the `summarize` operator unordered

### PR DESCRIPTION
**Pull request by dominiklohmann**
This makes the following pipeline 20x faster:

```
// tql2
load_http "https://storage.googleapis.com/tenzir-datasets/M57/suricata.json.zst"
decompress "zstd"
read_suricata
where @name == "suricata.flow"
summarize distinct(flow.state)
```

### CODE REVIEW

File: libtenzir/builtins/operators/summarize.cpp [hunk 1]
```diff
@@ -1132,8 +1132,9 @@ class summarize_operator2 final : public crtp_operator<summarize_operator2> {
 
   auto optimize(expression const& filter, event_order order) const
     -> optimize_result override {
-    TENZIR_UNUSED(filter, order);
-    return do_not_optimize(*this);
+    // Note: The `unordered` relies on commutativity of the aggregation functions.
```
Comment (mavam) [inline libtenzir/builtins/operators/summarize.cpp:1135]: Well, kinda. If you `collect` you may get `[1,2]` or `[2,1]`. From a set perspective, this is commutative, but not from a list/array perspective.

Comment (mavam) [reply libtenzir/builtins/operators/summarize.cpp:1135]: I think associativity is really what's required here, which basically means "it doesn't matter how you pair things up".

**Review by mavam — APPROVED** [review]

File: libtenzir/builtins/operators/summarize.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libtenzir/builtins/operators/summarize.cpp:1135]: I just copied this comment 1:1 from the TQL1 implementation, I didn't even read it 😅 Do you have a concrete suggestion? Also, should we perhaps explain this in the operator reference?

Comment (mavam) [reply libtenzir/builtins/operators/summarize.cpp:1135]: Just remove the comment to avoid confusion for now :-)

Comment (jachris) [reply libtenzir/builtins/operators/summarize.cpp]: I wrote that comment originally. Your `collect` example shows exactly why commutativity matters. Strictly speaking, we'd either have to document that `summarize` is free to reorder, or we have to disable the optimization if a non-commutative aggregation function is used. But maybe it's a bit pedantic and fine as it is now. Still wanted to clarify that the problem associated with this comment is real.

Comment (mavam) [reply libtenzir/builtins/operators/summarize.cpp]: > we'd either have to document that `summarize` is free to reorder,

This.

**Pull request merged**